.github/workflows/maestro_release.yaml

```diff
@@ -80,9 +79,56 @@ jobs:
         with:
           name: maestro-dist
           path: dist
+          
+      - uses: actions/checkout@v4
+      
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - name: Install uv and activate the environment
+        uses: astral-sh/setup-uv@v6
+        with:
+          activate-environment: true
+
+      - name: Install dependencies
+        run: |
+          uv pip install -e .[crewai]
+
+      - name: Get release name
+        id: get_name
+        run: python tools/get_release_name.py
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_REPOSITORY: ${{ github.repository }}
+
+      - name: Mock artifact for test release
+        run: |
+          mkdir -p dist && echo "test" > dist/fake.txt
+
       - name: GitHub Release
         uses: softprops/action-gh-release@v2
         if: github.ref_type == 'tag'
         with:
           files: dist/*
           generate_release_notes: true
+          name: ${{ steps.get_name.outputs.release_name }} – ${{ github.ref_name }}
+
+      - name: Strike through used release name
+        run: |
+          name="${{ steps.get_name.outputs.release_name }}"
+          pattern=$(printf '%s' "$name" | sed -e 's/[]\/$*.^[]/\\&/g')
+          sed -i "0,/- $pattern/s//- ~~$name~~/" .release_names.md
+          
+      - name: Create pull request
+        uses: peter-evans/create-pull-request@v6
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          commit-message: "chore: strike out used release name '${{ steps.get_name.outputs.release_name }}'"
+          branch: strikeout-${{ github.ref_name }}-${{ github.run_id }}
+          base: main
+          title: "chore: strike out used release name '${{ steps.get_name.outputs.release_name }}'"
+          body: |
+            This PR was automatically created after tagging `${{ github.ref_name }}` to strike through the used release name.
+          labels: automated-pr
```

Comment:
I would argue that this should be in its own followup job. In addition if we're already opening a followup to release PR then we should also have it increment the version in pyproject.toml and README.md to the next expected version (I suggested this a followup in the previous PR)

Reply:
if you open a followup then this part LGTM